packages/@react-aria/interactions/src/useFocusVisible.ts

```diff
@@ -106,7 +108,7 @@ function handleFocusEvent(e: FocusEvent) {
 
   // If a focus event occurs without a preceding keyboard or pointer event, switch to virtual modality.
   // This occurs, for example, when navigating a form with the next/previous buttons on iOS.
-  if (!hasEventBeforeFocus && !hasChangedTabRecently()) {
+  if (!hasEventBeforeFocus && !hasBlurredWindowRecently && !hasChangedTabRecently()) {
```

Comment:
Is `hasChangedTabRecently` still needed then? I assume the blur event is also fired when switching tabs?

Reply:
Actually, yes! I have checked it for all browsers and it works fine, will adjust the solution 👍